### PR TITLE
iscsi volume ConstructVolumeSpec with right content

### PR DESCRIPTION
**Pull request by wenjun93**
**What this PR does / why we need it**:
fix ConstructVolumeSpec in iscsi volume plugin, 
iscsi plugin.ConstructVolumeSpec() construct volume spec with fake value, cause kubelet volume manager will create two volumesInUse in node Status.

```
**before fix:** 
volumesAttached:
  - devicePath: ""
    name: kubernetes.io/iscsi/xx.xxx.xxx.xx:3260:iqn.2010-01.com.xxx.uuid-699ea5f8-f9de-4f26-bcfc-89135c5de715.831681:0
  volumesInUse:
  - kubernetes.io/iscsi/pvc-447a4519-daf6-11e7-83c0-90e2bab4ed80:pvc-447a4519-daf6-11e7-83c0-90e2bab4ed80:0
  - kubernetes.io/iscsi/xx.xxx.xxx.xx:3260:iqn.2010-01.com.xxx.uuid-699ea5f8-f9de-4f26-bcfc-89135c5de715.831681:0

**after fix of this PR:**
 volumesAttached:
  - devicePath: ""
    name: kubernetes.io/iscsi/xx.xxx.xxx.xx:3260:iqn.2010-01.com.xxx.uuid-699ea5f8-f9de-4f26-bcfc-89135c5de715.831681:0
  volumesInUse:
  - kubernetes.io/iscsi/xx.xxx.xxx.xx:3260:iqn.2010-01.com.xxx.uuid-699ea5f8-f9de-4f26-bcfc-89135c5de715.831681:0

volumesInUse now align with volumesAttached.

### CODE REVIEW

**Comment by wenjun93** [comment]
@jsafrane any ideas on this change please? thanks!

**Comment by dims** [comment]
cc @smcginnis

**Comment by rootfs** [comment]
@wenjun93 see https://github.com/kubernetes/kubernetes/issues/57744

File: pkg/volume/iscsi/iscsi.go [hunk 1]
```diff
@@ -213,12 +213,29 @@ func (plugin *iscsiPlugin) newUnmounterInternal(volName string, podUID types.UID
...
 
 func (plugin *iscsiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.Spec, error) {
+	mounter := plugin.host.GetMounter(plugin.GetPluginName())
+	pluginDir := plugin.host.GetPluginDir(plugin.GetPluginName())
+	sourceName, err := mounter.GetDeviceNameFromMount(mountPath, pluginDir)
+	if err != nil {
+		return nil, err
+	}
+	str := strings.Split(sourceName, "-lun-")
+	index := strings.Index(str[0], "-")
```
Comment (rootfs) [inline pkg/volume/iscsi/iscsi.go]: validate str array first before use

Comment (wenjun93) [reply pkg/volume/iscsi/iscsi.go]: @rootfs I put it in line below(L224), if str is empty or not len(2), will return error, str[0] (L223) will not cause panic here.

**Comment by wenjun93** [comment]
with #57744 , need to rewrite

**Comment by jsafrane** [comment]
And also please add some unit tests to check it works as expected.

**Comment by rootfs** [comment]
cc @mtanino  #58089

**Comment by mtanino** [comment]
@wenjun93
Thank you for the PR. Yesterday, iSCSI Block Volume support update was merged, update of ConstructVolumeSpec() also got merged together in the update.
Could you check latest code and try whether your issue is fixed or not?

**Comment by mtanino** [comment]
FYI:
Block volumes Support: iSCSI plugin update #54752
https://github.com/kubernetes/kubernetes/pull/54752

**Comment by wenjun93** [comment]
@mtanino hi, I've checked your change, thank you very much

**Pull request closed without merging**